tests/utils/test_agent_tools.py

```
@@ -0,0 +1,788 @@
+            observed=peer2.name,
+            session_name=session.name,
+        )
+
+        # Call with missing required parameter - should return error string
+        result = await executor("search_memory", {})  # Missing 'query'
+
+        assert isinstance(result, str)
+        # Should contain error info, not raise exception
+
+    async def test_executor_dreamer_context_includes_observation_ids(
+        self, db_session: AsyncSession, tool_test_data: Any
```

Ruff (0.14.10)

Dynamically typed expressions (typing.Any) are disallowed in `tool_test_data`

(ANN401)